src/Tools/ConfigurationSchemaGenerator/ConfigSchemaEmitter.cs

```diff
@@ -719,6 +715,22 @@ private static string[] CreateExclusionPaths(List<string>? exclusionPaths)
         return result;
     }
 
+    private static void ReplaceNodeWithKeyCasingChange(JsonObject jsonObject, string key, JsonNode value)
+    {
+        // In System.Text.Json v9, the casing of the new key is not adapted. See https://github.com/dotnet/runtime/issues/108790.
+        // So instead, remove the existing node and insert a new one with the updated key.
+        var index = jsonObject.IndexOf(key);
```

Comment:
This workaround requires dual lookup if the key is missing, which suggests to me that a `JsonObject.TryAdd`method similar to the one in https://github.com/dotnet/runtime/issues/107947 might be necessary.